server/src/food/handler/FoodItem.handler.ts

```diff
@@ -0,0 +1,127 @@
+import { NextFunction, Request, Response } from "express";
+import { Types } from "mongoose";
+import { DbError, ERRORS } from "../../app/index";
+import { DbExtension } from "../../db/interface/db-extension";
+import { FoodItem } from "../index";
+
+export class FoodItemHandler {
+
+  constructor (private _db: DbExtension) {}
+
+  public getAll = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      let result;
+
+      if (req.query.hasOwnProperty("socobouserid")) {
+        const socoboUserId = req.query.socobouserid;
+        await this._checkIfSocoboUserExists(socoboUserId, "getAll(..)");
+
+        result = (req.query.hasOwnProperty("resolve"))
+          ? result = await this._db.fooditem.getAllBySocoboUserIdResolved(socoboUserId, false)
+          : (req.query.hasOwnProperty("resolve-deep"))
+            ? await this._db.fooditem.getAllBySocoboUserIdResolved(socoboUserId, true)
+            : await this._db.fooditem.getAllBySocoboUserId(socoboUserId);
+
+      } else {
+
+        result = (req.query.hasOwnProperty("resolve"))
+          ? await this._db.fooditem.getAllResolved(false)
+          : (req.query.hasOwnProperty("resolve-deep"))
+            ? await this._db.fooditem.getAllResolved(true)
+            : await this._db.fooditem.getAll();
+      }
+
+      res.status(200).json(result);
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  public getById = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+
+      const id = new Types.ObjectId(req.params.id);
+      const result = (req.query.hasOwnProperty("resolve"))
+        ? await this._db.fooditem.getByIdResolved(id, false)
+        : (req.query.hasOwnProperty("resolve-deep"))
+          ? await this._db.fooditem.getByIdResolved(id, true)
+          : await this._db.fooditem.getById(id);
+
+      res.status(200).json(result);
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  public save = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
```

Comment:
There was an error for me with the POST:

Request:

{
        "foodItemTemplateId": "59a2eea98a5a150e9e829409",
        "socoboUserId": "59a2ee598ae9b7b45243d503",
        "amount": 3333,
        "bestBefore": 1501948817400
    }

Error

{
    "cause": "item.socoboUserId.toHexString is not a function",
    "message": "Internal server error",
    "method": "",
    "source": "",
    "type": "TypeError"
}

Reply:
I will check this tomorrow I guess.